components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java

```diff
@@ -0,0 +1,172 @@
+                            LOG.error("Error while proceeding after successful response from Choreo call for "
+                                    + "session data key: " + authenticationContext.getContextIdentifier(), e);
+                        }
+                    }
+
+                    @Override
+                    public void failed(final Exception ex) {
+
+                        LOG.error("Failed to invoke Choreo for session data key: " +
+                                authenticationContext.getContextIdentifier(), ex);
+                        try {
+                            if (LOG.isDebugEnabled()) {
+                                LOG.debug(" Calls to Choreo failed for session " +
+                                        "data key: " + authenticationContext.getContextIdentifier());
+                            }
+                            String outcome = OUTCOME_FAIL;
+                            if ((ex instanceof SocketTimeoutException)
+                                    || (ex instanceof ConnectTimeoutException)) {
+                                outcome = OUTCOME_TIMEOUT;
+                            }
+                            asyncReturn.accept(authenticationContext, Collections.emptyMap(), outcome);
+                        } catch (FrameworkException e) {
+                            LOG.error("Error while proceeding after failed response from Choreo " +
+                                    "call for session data key: " + authenticationContext
+                                    .getContextIdentifier(), e);
+                        }
+                    }
+
+                    @Override
+                    public void cancelled() {
+
+                        LOG.error("Invocation Choreo for session data key: " +
+                                authenticationContext.getContextIdentifier() + " is cancelled.");
+                        try {
+                            if (LOG.isDebugEnabled()) {
+                                LOG.debug(" Calls to Choreo for session data key: "
+                                        + authenticationContext.getContextIdentifier());
+                            }
+                            asyncReturn.accept(authenticationContext, Collections.emptyMap(), OUTCOME_FAIL);
+                        } catch (FrameworkException e) {
+                            LOG.error("Error while proceeding after cancelled response from Choreo " +
+                                    "call for session data key: " + authenticationContext
+                                    .getContextIdentifier(), e);
+                        }
+                    }
+                });
+            } catch (IllegalArgumentException e) {
+                LOG.error("Invalid endpoint Url: ", e);
+                asyncReturn.accept(authenticationContext, Collections.emptyMap(), OUTCOME_FAIL);
+            } catch (IOException e) {
+                LOG.error("Error while calling endpoint. ", e);
```

Comment:
Extra space at the end

Reply:
Addressed with commit [dfdefae](https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/85/commits/dfdefae64b30f62bb470af184fd87d775ffdfb2d)